compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts

```diff
@@ -194,19 +209,15 @@ function pushPropertyLoadNode(
     !isMutableAtInstr ||
     knownImmutableIdentifiers.has(node.fullPath.identifier.id)
   ) {
-    let curr: PropertyLoadNode | null = node;
-    while (curr != null) {
-      result.add(curr);
-      curr = curr.parent;
-    }
+    result.add(node);
```

Comment:
Trust that all previous property loads were added correctly. If a property load chain ever spans multiple blocks, this is technically the correct behavior